packit_service/worker/build/build_helper.py

```diff
@@ -323,10 +324,34 @@ def get_test_check(cls, chroot: str = None) -> str:
         chroot_str = f"-{chroot}" if chroot else ""
         return f"{deployment_str}/{cls.status_name_test}{chroot_str}"
 
-    def create_srpm_if_needed(self) -> None:
-        """If you want to be sure we already created the SRPM."""
-        if not (self._srpm_path or self._srpm_model):
-            self._create_srpm()
+    def create_srpm_if_needed(self) -> Optional[TaskResults]:
+        """
+        Create SRPM if is needed.
+
+        Returns:
+            Task results if job is cancelled because of merge conflicts, `None`
+        otherwise.
```

Comment:
This can be a bit unclear for someone. But I am not strictly against it. (I see how it makes the code shorter.)